### PR TITLE
Email notifications subscription to Incident updates

### PR DESCRIPTION
**Pull request by pdcp1**
This PR is related with issue #982 

image

- Adds a `Notify Me of Updates` button on the incident page
  - If the user is not logged in, it prompts the user to log in before subscribing
  - If the user is logged in, it subscribes the user to receive email notifications each time that a new report is associated to that incident
 
- Adds a new collection `subscriptions` on `customData` DB

- Sends an email to each subscriber when a new report is associated with a specific incident
  - It uses Sendgrid (`@sendgrid/mail` package) to send the email
  - The email template is on`site/gatsby-site/src/emailTemplates/incidentUpdated.html`
 
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -584,6 +584,12 @@ To get your Public and Private API Key, follow these [instructions](https://www.
 To get the group ID and the app ID, the easiest way is to navigate to your Atlas Service App dashboard and copy from the URL.
 The URL format is https://realm.mongodb.com/groups/[groupId]/apps/[appId]/dashboard
 
+In addition to that, this value should be added to the `.env` file:
```
Comment (cesarvarela) [inline README.md:587]: Maybe specify that this is for development only?

File: site/gatsby-site/src/contexts/userContext/UserContext.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ export const UserContext = createContext({
   user: undefined,
   isAdmin: undefined,
   isLoggedIn: undefined,
-  isRole() {
+  isRole(role) {
+    role;
```
Comment (cesarvarela) [inline site/gatsby-site/src/contexts/userContext/UserContext.js:9]: this is weird but I get why you did it 🤷

File: site/gatsby-site/src/templates/cite.js [hunk 3]
```diff
@@ -1,33 +1,37 @@
...
+import { getCanonicalUrl } from '../utils/getCanonicalUrl';
 import { format, isAfter, isEqual } from 'date-fns';
 import { useModal, CustomModal } from '../hooks/useModal';
-import Timeline from 'components/visualizations/Timeline';
-import IncidentStatsCard from 'components/cite/IncidentStatsCard';
-import IncidentCard from 'components/cite/IncidentCard';
-import Taxonomy from 'components/taxa/Taxonomy';
-import { useUserContext } from 'contexts/userContext';
-import SimilarIncidents from 'components/cite/SimilarIncidents';
+import Timeline from '../components/visualizations/Timeline';
```
Comment (cesarvarela) [inline site/gatsby-site/src/templates/cite.js:11]: Please undo these changes, it throws away the benefits of using the webpack mappings from:  https://github.com/responsible-ai-collaborative/aiid/blob/master/site/gatsby-site/gatsby-node.js#L71

File: site/gatsby-site/src/templates/cite.js [hunk 4]
```diff
@@ -141,6 +147,59 @@ function CitePage(props) {
...
+    useMutation(UPSERT_SUBSCRIPTION);
+
+  const subscribeToNewReports = async () => {
+    if (isRole('subscriber')) {
+      try {
+        await subscribeToNewReportsMutation({
+          variables: {
+            query: {
+              type: 'incident',
+              userId: { userId: user.id },
```
Comment (cesarvarela) [inline site/gatsby-site/src/templates/cite.js:160]: I think this enables anyone to subscribe other users?

I wonder if there is something like subscribe "whoever is sending this request" to prevent that from happening.

**Review by cesarvarela — CHANGES_REQUESTED** [review]

File: site/gatsby-site/src/contexts/userContext/UserContext.js (hunk 2 above)
Comment (pdcp1) [reply site/gatsby-site/src/contexts/userContext/UserContext.js:9]: It's to avoid eslint errors 🤷

File: site/realm/functions/getUser.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            }
+
+        }
+        // The response body is a BSON.Binary object. Parse it and return
+        return EJSON.parse(userResponse.body.text()).data;
+
+    }
+    catch (e) {
+        if (e.response) {
+            return {
```
[the remark this answers is not in the document]
Comment (pdcp1) [reply site/realm/functions/getUser.js:49]: This is to return the error code and error message in a simpler and cleaner way, trying to avoid returning the full exception as a response

File: README.md (hunk 1 above)
Comment (pdcp1) [reply README.md:587]: Actually, after the refactor to move the sendEmail code to a function, this `SENDGRID_API_KEY` value is not needed

File: site/gatsby-site/src/templates/cite.js (hunk 3 above)
Comment (pdcp1) [reply site/gatsby-site/src/templates/cite.js:11]: Done

File: site/realm/functions/getUser.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (cesarvarela) [reply site/realm/functions/getUser.js:49]: Sounds good, this will be visible on Atlas' logs right?

File: site/realm/data_sources/mongodb-atlas/customData/users/rules.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+    "collection": "users",
+    "database": "customData",
+    "roles": [
+        {
+            "name": "readAll",
```
[the remark this answers is not in the document]
Comment (pdcp1) [reply site/realm/data_sources/mongodb-atlas/customData/users/rules.json:6]: done!

**Comment by pdcp1** [comment]
@cesarvarela all changes are done and ready to review and test on https://deploy-preview-1002--staging-aiid.netlify.app

**Comment by pdcp1** [comment]
@cesarvarela this is the link to test https://deploy-preview-4--pablocosta-staging.netlify.app

**Review by cesarvarela — APPROVED** [review]
